src/main/groovy/com/ceilfors/jenkins/plugins/jiratrigger/JiraTrigger.groovy

```diff
@@ -128,8 +128,20 @@ abstract class JiraTrigger<T> extends Trigger<Job> {
         }
 
         protected void addTrigger(JiraTrigger jiraTrigger) {
+            def addOrUpdate = "Added"
+            triggers.removeIf {
+                if (it.job.fullName == jiraTrigger.job.fullName &&
```

Comment:
Would this remove a trigger even when the 2 different JIRA issue is getting updated? Let's say Issue A in JIRA gets updated, and Issue B gets updated very quickly, will it remove Issue A job trigger?

Reply:
Hi, Tanasa San, addTrigger is called not when an issue is updated, but when a job that is to be triggered is added/updated.

When a issue is updated, the trigger (i.e., the job) is triggered, or fired, and it is run (but not added, these are different things).

Here's how to reproduce the bug that I'm talking about. It is kinda related to what you are worrying about, except instead of issue A and B firing quickly, it happens when only ONE  issue A is triggered very quickly twice, I would like the job to be built exactly twice.

But I found that the job is built only once, i.e., Jenkins would quietly merge them. So I added a Random String Parameter, that exactly prevented builds from being merged.

Now, to my dismay, I found out that even if the issue is updated only once, the job could be scheduled to be run multiple times.

Later, I found out the relationship between the number of times the job is run for a single update, and the number of times I updated the job (when I am developing the IssueMatcher to replace the jqlFilter).

Hence this patch.